Source/WTF/wtf/text/StringCommon.h

```diff
@@ -813,6 +845,63 @@ template<typename StringClass> bool equalLettersIgnoringASCIICaseCommon(const St
     return hasPrefixWithLettersIgnoringASCIICaseCommon(string, literal);
 }
 
+template <typename SearchCharacterType, typename MatchCharacterType>
+bool startsWith(std::span<const SearchCharacterType> string, std::span<const MatchCharacterType> prefix)
+{
+    if (prefix.size() > string.size())
+        return false;
+
+    return equal(string.data(), prefix);
+}
+
+template <typename SearchCharacterType, typename MatchCharacterType>
+bool endsWith(std::span<const SearchCharacterType> string, std::span<const MatchCharacterType> suffix)
+{
+    unsigned suffixSize = suffix.size();
+    unsigned referenceSize = string.size();
+    if (suffixSize > referenceSize)
+        return false;
+
+    unsigned startOffset = referenceSize - suffixSize;
+
+    return equal(string.subspan(startOffset).data(), suffix);
+}
+
+template <typename SearchCharacterType, typename MatchCharacterType>
+bool endsWithLettersIgnoringASCIICaseCommon(std::span<const SearchCharacterType> string, std::span<const MatchCharacterType> suffix)
+{
+    unsigned suffixLength = suffix.size();
+    unsigned referenceLength = string.size();
+    if (suffixLength > referenceLength)
+        return false;
+
+    unsigned startOffset = referenceLength - suffixLength;
+
+    return equalIgnoringASCIICaseWithLength(string.subspan(startOffset), suffix, suffixLength);
+}
+
+template <typename SearchCharacterType, typename MatchCharacterType>
+bool endsWithLettersIgnoringASCIICase(std::span<const SearchCharacterType> string, std::span<const MatchCharacterType> suffix)
```

Comment:
It doesn’t make sense to use a template and `MatchCharacterType` here. The "letters" functions requires an ASCII string for the suffix, so it should always be the same type and not a template. I think, in fact, we might want to switch to use char instead of Latin1Character for the suffix arguments.